File: 프로그래머스/0/120895. 인덱스 바꾸기/인덱스 바꾸기.java
// 250723

class Solution {
    public String solution(String my_string, int num1, int num2) {
        String answer = "";
        
        StringBuilder sb = new StringBuilder(my_string);
        
        char tmp = sb.charAt(num1);
        sb.setCharAt(num1, sb.charAt(num2));
        sb.setCharAt(num2, tmp);
        
        answer += sb.toString();
        
        return answer;
    }
}

File: 프로그래머스/0/120895. 인덱스 바꾸기/인덱스 바꾸기.py
# 260202

def solution(my_string, num1, num2):
    answer = ''
    
    # num1, num2 순서 고정
    if num1 > num2:
        tmp = num1
        num1 = num2
        num2 = tmp
    
    for i in range(len(my_string)):
        if i == num1:
            answer += my_string[num2]
        elif i == num2:
            answer += my_string[num1]
        else:
            answer += my_string[i]
    
    return answer